File: src/go/t6/s1.go
package main

import "strconv"

func Fib(number int) int  {
	if number < 0 {
		return -1
	}

	if number <= 1 {
		return number
	}

	return Fib(number-1) + Fib(number-2)
}

func Fibonacci(number int) (int, error) {
	result := Fib(number)
	s := strconv.Itoa(result)

	if len(s) > 6 {
		return strconv.Atoi(s[len(s)-6:])
	}

	return strconv.Atoi(s)
}

File: src/php/t6/s1.php
<?php

declare(strict_types=1);

function fib(int $n): int {
    if ($n < 0) {
        throw new Exception("Only non-negative integers are allowed");
    }

    if($n <= 1) {
        return $n;
    }

    return fibonacci($n-1) + fibonacci($n-2);
}

function fibonacci(int $n): int {
    $result = fib($n);

    $str = (string) $result;
    if (strlen($str) > 6 ) {
        return (int) substr($str, -6);
    }

    return (int) $str;
}
